### PR TITLE
Support managed rowcopy

### PR DESCRIPTION
**Pull request by everpcpc**
Related issue: https://github.com/github/gh-ost/pull/91 https://github.com/github/gh-ost/issues/417

### Description

This PR adds support for managed rowcopy.

### CODE REVIEW

**Comment by shlomi-noach** [comment]
Thank you! I'll be reviewing this PR in the next few days.

**Comment by shlomi-noach** [comment]
Please let me know when this will be ready for review (for sure I won't begin till next week)

**Comment by everpcpc** [comment]
It's ready for review now~

**Comment by everpcpc** [comment]
To be more clear, I removed other changes and only left the managed rowcopy part.
It would be easier for review now~

**Comment by shlomi-noach** [comment]
Thank you! It's now a matter of setting serious time aside to review.

File: go/logic/applier.go [hunk 1]
```diff
@@ -459,6 +460,119 @@ func (this *Applier) CalculateNextIterationRangeEndValues() (hasFurtherRange boo
...
+				return err
+			}
+			for rows.Next() {
+				err = rows.Scan(scanArgs...)
+				if err != nil {
+					return err
+				}
+				applyQuery += applyPreparedQuery
+				for _, value := range values {
+					if value == nil {
```
Comment (MOON-CLJ) [inline go/logic/applier.go]: if dont convert to string, no need to distinguish value is or not nil?

**Pull request closed without merging**